server/sonar-server/src/main/java/org/sonar/server/computation/scm/ScmInfoRepository.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.scm;
+
+import com.google.common.base.Optional;
+import org.sonar.server.computation.component.Component;
+
+/**
+ * Return SCM information of components.
+ *
+ * It will always search in the db if there's nothing in the report.
+ */
+public interface ScmInfoRepository {
+
+  /**
+   * Return SCM information for the given component.
+   * First it try to find them in the report, and if nothing is found then it will search them in the database.
```

Comment:
"Will return Scm info from the report if there is any, otherwise will look them up in the DB. If there is no SCM info in the database, returns Optional#absent()"